client/ayon_applications/utils.py

```diff
@@ -453,10 +453,12 @@ def prepare_app_environments(
     # Add tools environments
     groups_by_name = {}
     tool_by_group_name = collections.defaultdict(dict)
+    used_tool_names = set()
     for key in tools:
         tool = app.manager.tools.get(key)
         if not tool or not tool.is_valid_for_app(app):
             continue
+        used_tool_names.add(tool.full_name)
```

Comment:
Are tools technically applied in an order? If so, would it be worth to keep this a list to be entirely accurate, or?